spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/trace/sleuth/SpanTranslator.java

```diff
@@ -47,6 +47,8 @@
 		Map<Span.Kind, SpanKind> map = new HashMap<>();
 		map.put(Span.Kind.CLIENT, SpanKind.RPC_CLIENT);
 		map.put(Span.Kind.SERVER, SpanKind.RPC_SERVER);
+		map.put(Span.Kind.CONSUMER, SpanKind.RPC_CLIENT);
+		map.put(Span.Kind.PRODUCER, SpanKind.RPC_SERVER);
```

Comment:
Is there some guidance what and how to treat?
Why did you do this choice?
I would say that `producer` is the client, and `consumer` is a server...

Thanks

Reply:
SpanKind only has RPC_CLIENT, RPC_SERVER, UNSPECIFIED and UNKNOWN, the latter one errors out.
I thought this was the most intuitive mapping. A server produces data and a client consumes. Although I guess it could really be the other way around.

Should both these mappings be set as UNSPECIFIED then?

Reply:
We can file an issue with google-cloud-java to add a new category. But it's not guaranteed that they'll add it.

For now, based on the following line, I'm leaning to make these span kind `UNSPECIFIED`:

```Unlike {@link #SERVER}, messaging spans never share a span ID. For example, the {@link
     * #PRODUCER} of this message is the {@link TraceContext#parentId()} of this span.```

The distinction between client and server in the google-cloud side seems to be just for display purposes.

https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#SpanKind

@artembilan @meltsufin wdyt?

Reply:
I'm fine with `UNSPECIFIED`. At least we fix the issue with non-mapped type. Plus this is going to be some middle ground until the final decision with the Google Cloud Java is arranged.